pulpcore/app/models/publication.py

```diff
@@ -553,3 +556,36 @@ def invalidate_cache(self):
         if settings.CACHE_ENABLED:
             Cache().delete(base_key=self.base_path)
             # Can also preload cache here possibly
+
+
+class ArtifactDistribution(Distribution):
+    """Serve artifacts by their uuid.
+    This is in tech-preview.
+    """
+
+    TYPE = "artifact"
+
+    def artifact_url(self, artifact):
+        origin = settings.CONTENT_ORIGIN.strip("/")
+        prefix = settings.CONTENT_PATH_PREFIX.strip("/")
+        base_path = self.base_path.strip("/")
+
+        plain_url = urljoin(
+            urljoin(urljoin(origin, prefix + "/"), base_path + "/"), str(artifact.pk)
+        )
+        return self.content_guard.cast().preauthenticate_url(plain_url)
+
+    def content_handler(self, path):
+        """Serve an artifact or return 404."""
+        uuid = path.rstrip("/")
+        if re.match(r"[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}", uuid):
+            artifact = Artifact.objects.get(pk=uuid)
```

Comment:
use filter() instead or if with get() then you need to catch doesnotexist error

Reply:
```suggestion
            artifact = Artifact.objects.first(pk=uuid)
```
Is probably a good solution.

Reply:
first() does not take any args/kwargs, it should be filter()

Reply:
and `filter` returns a query set. So `filter(pk=uuid).first()` ?

Reply:
yeah that will work